tracer/src/Datadog.Trace/Debugger/DebuggerManager.cs

```diff
@@ -152,203 +298,304 @@ private void SetExceptionReplayState()
+
+            if (disabled)
             {
-                // Handle cancellation gracefully
+                TracerManager.Instance.Telemetry.ProductChanged(TelemetryProductType.DynamicInstrumentation, enabled: false, error: null);
             }
-            catch (Exception ex)
+
+            if (dynamicallyDisabled)
             {
-                Log.Error(ex, "Error updating debugger state");
+                Log.Debug("Dynamic Instrumentation is disabled by remote enablement. To enable it, re-enable it via Datadog UI");
             }
-            finally
+            else
             {
-                if (semaphoreAcquired)
-                {
-                    try
-                    {
-                        _semaphore.Release();
-                    }
-                    catch
-                    {
-                        // ignore ObjectDisposedException or SemaphoreFullException
-                    }
-                }
+                Log.Debug("Dynamic Instrumentation is disabled. To enable it, please set {DynamicInstrumentationEnabled} environment variable to 'true'.", ConfigurationKeys.Debugger.DynamicInstrumentationEnabled);
             }
         }
 
-        private void WriteDebuggerDiagnosticLog()
+        private void WriteStartupDebuggerDiagnosticLog()
         {
             try
             {
                 var stringWriter = new StringWriter();
+                var settings = DebuggerSettings;
                 using (var writer = new JsonTextWriter(stringWriter))
                 {
-                    var debuggerSettings = DebuggerSettings;
+                    writer.WriteStartObject();
                     writer.WritePropertyName("dynamic_instrumentation_enabled");
-                    writer.WriteValue(debuggerSettings.DynamicInstrumentationEnabled);
-                    writer.WritePropertyName("symbol_database_upload_enabled");
-                    writer.WriteValue(debuggerSettings.SymbolDatabaseUploadEnabled);
+                    writer.WriteValue(settings.DynamicInstrumentationEnabled);
                     writer.WritePropertyName("code_origin_for_spans_enabled");
-                    writer.WriteValue(debuggerSettings.CodeOriginForSpansEnabled);
-                    writer.WritePropertyName("symbol_database_upload_enabled");
-                    writer.WriteValue(DebuggerSettings.SymbolDatabaseUploadEnabled);
+                    writer.WriteValue(settings.CodeOriginForSpansEnabled);
                     writer.WritePropertyName("exception_replay_enabled");
                     writer.WriteValue(ExceptionReplaySettings.Enabled);
+                    writer.WritePropertyName("symbol_database_upload_enabled");
+                    writer.WriteValue(settings.SymbolDatabaseUploadEnabled);
+                    writer.WriteEndObject();
                 }
 
                 Log.Information("DATADOG DEBUGGER CONFIGURATION - {Configuration}", stringWriter.ToString());
```

Comment:
Given you always write this log (which I think is fine, as it mirrors the tracer version 👍), you can probably change the other logs that are used by tests to use this one, instead of writing additional logs